compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -2919,4 +3006,295 @@ mod test {
             }
         }
     }
+
+    // Test that given multiple primitive operations that are represented by Brillig directives (e.g. invert/quotient),
+    // we will only generate one bytecode and the appropriate Brillig call opcodes are generated.
+    #[test]
+    fn multiple_brillig_stdlib_calls() {
+        // acir(inline) fn main f0 {
+        //     b0(v0: u32, v1: u32, v2: u32):
+        //       v3 = div v0, v1
+        //       constrain v3 == v2
+        //       v4 = div v2, v2
+        //       constrain v4 == u32 1
```

Comment:
Note that these instructions are going to be immediately optimized out as you're adding them to the circuit so this test isn't actually testing what we want.

We should probably change this to be `div v1, v2`

Reply:
Similarly in the other two tests.

Reply:
> the SSA knows that v2/v2 -> 1 no matter the value of v2

On second thoughts, perhaps this should have a zero check on the rhs? Opening an issue.

Reply:
As I am not running optimizations the instruction is still there. But I did think of the same thing so I will change it to be safe

Reply:
We automatically simplify instructions as they're inserted so we don't need to run an explicit optimisation pass. (The constant folding pass is just a matter of reinserting all the instructions one-by-one)

Reply:
ah yes you are correct, but the test still passed as the instruction was not optimized out

Reply:
I switched to `div v1, v2`